diff --git a/examples/bc_image_real.py b/examples/bc_image_real.py
index 7c66734..5e923da 100644
--- a/examples/bc_image_real.py
+++ b/examples/bc_image_real.py
@@ -108,6 +108,13 @@ def experiment(variant):
     elif args.buffer == 4:
         print('Stephen Tool Use')
         path = '/nfs/kun1/users/stephentian/on_policy_longer_1_26_buffers/move_tool_obj_together_fixed_6_2_train.pkl'
+    elif args.buffer == 5:
+        print('Albert Pick Place')
+        px = os.path.join(os.path.expanduser("~"),'val_data_relabeled', 'combined_2021-06-03_21_36_48_labeled.pkl') if args.azure else '/nfs/kun1/users/albert/realrobot_datasets/combined_2021-06-03_21_36_48_labeled.pkl'
+        data_path = '/nfs/kun1/users/albert/realrobot_datasets/combined_2021-06-03_21_36_48_labeled.pkl'
+        if args.azure:
+            data_path = px
+        paths.append((data_path, None))
     else:
         assert False
     
@@ -253,7 +260,7 @@ if __name__ == "__main__":
     parser.add_argument('--vqvae_enc', action="store_true", default=False)
     parser.add_argument('--duplicate', action="store_true", default=False)
     parser.add_argument('--num_traj', default=0, type=int)
-    parser.add_argument('--smimg', default=True, action='store_false')
+    parser.add_argument('--smimg', default=False, action='store_true')
     parser.add_argument('--imgstate', default=False, action='store_true') # both image and state
     parser.add_argument('--state_dim', default=3, type=int)
 
